standalone/src/test/java/io/pravega/local/AuthEnabledInProcPravegaClusterTest.java

```diff
@@ -0,0 +1,161 @@
+     *
+     */
+    @Test(timeout = 50000)
+    public void testWriteAndReadEventsWhenConfigIsProper() throws ReinitializationRequiredException {
+        String scope = "org.example.auth";
+        String streamName = "stream1";
+        int numSegments = 1;
+
+        ClientConfig clientConfig = ClientConfig.builder()
+                .credentials(new DefaultCredentials("1111_aaaa", "admin"))
+                .controllerURI(URI.create(localPravega.getInProcPravegaCluster().getControllerURI()))
+                .build();
+        log.info("Done creating client config");
+
+        @Cleanup
+        StreamManager streamManager = StreamManager.create(clientConfig);
+        log.info("Created a stream manager");
+
+        streamManager.createScope(scope);
+        log.info("Created a scope [{}]", scope);
+
+        streamManager.createStream(scope, streamName, StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(numSegments))
+                .build());
+        log.info("Created a stream with name [{}]", streamName);
+
+        @Cleanup
+        ClientFactory clientFactory = ClientFactory.withScope(scope, clientConfig);
+
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName,
+                new JavaSerializer<String>(),
+                EventWriterConfig.builder().build());
+        log.info("Got a writer");
+
+        String writeEvent1 = "This is event 1";
+        writer.writeEvent(writeEvent1);
+        log.info("Done writing event [{}]", writeEvent1);
+
+        String writeEvent2 = "This is event 2";
+        writer.writeEvent(writeEvent2);
+        log.info("Done writing event [{}]", writeEvent2);
+
+        // Now, read the events from the stream.
+
+        String readerGroup = UUID.randomUUID().toString().replace("-", "");
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                .stream(Stream.of(scope, streamName))
+                .disableAutomaticCheckpoints()
+                .build();
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, clientConfig);
+        readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig);
+
+        @Cleanup
+        EventStreamReader<String> reader = clientFactory.createReader(
+                "readerId", readerGroup,
+                new JavaSerializer<String>(), ReaderConfig.builder().build());
+
+        // Keeping the read timeout large so that there is ample time for reading the event even in
+        // case of abnormal delays in test environments.
+        String readEvent1 = reader.readNextEvent(10000).getEvent();
+        log.info("Done reading event [{}]", readEvent1);
+
+        String readEvent2 = reader.readNextEvent(10000).getEvent();
+        log.info("Done reading event [{}]", readEvent2);
+
+        assertEquals(writeEvent1, readEvent1);
```

Comment:
Can we get a checkpoint event here?

Reply:
There shouldn't be an occurrence of a checkpoint event here, `disableAutomaticCheckpoints()` is invoked on `readerGroupConfig` in line 115:

https://github.com/pravega/pravega/blob/f5ba3f3139563fd7da5cc58babccb2ae4f77da28/standalone/src/test/java/io/pravega/local/AuthEnabledInProcPravegaClusterTest.java#L113-L116